databricks/koalas/tests/test_indexes.py

```diff
@@ -82,12 +82,7 @@ def test_to_series(self):
         pidx = self.pdf.set_index('b', append=True).index
         kidx = self.kdf.set_index('b', append=True).index
 
-        if LooseVersion(pyspark.__version__) < LooseVersion('2.4'):
-            # PySpark < 2.4 does not support struct type with arrow enabled.
-            with self.sql_conf({'spark.sql.execution.arrow.enabled': False}):
-                self.assert_eq(kidx.to_series(), pidx.to_series())
-                self.assert_eq(kidx.to_series(name='a'), pidx.to_series(name='a'))
-        else:
+        with self.sql_conf({'spark.sql.execution.arrow.enabled': False}):
```

Comment:
Struct type is supported from Spark 3.0. It failed in 2.3 because fallback isn't available there.

In Spark 3.0, it's supported when Arrow optimization is enabled. They produce different results:


```
E   Left:
E      b
E   0  4    {'__index_level_0__': 0, 'b': 4}
E   1  5    {'__index_level_0__': 1, 'b': 5}
E   3  6    {'__index_level_0__': 3, 'b': 6}
E   5  3    {'__index_level_0__': 5, 'b': 3}
E   6  2    {'__index_level_0__': 6, 'b': 2}
E   8  1    {'__index_level_0__': 8, 'b': 1}
E   9  0    {'__index_level_0__': 9, 'b': 0}
E      0    {'__index_level_0__': 9, 'b': 0}
E      0    {'__index_level_0__': 9, 'b': 0}
E   dtype: object
E   object
E   
E   Right:
E      b
E   0  4    (0, 4)
E   1  5    (1, 5)
E   3  6    (3, 6)
E   5  3    (5, 3)
E   6  2    (6, 2)
E   8  1    (8, 1)
E   9  0    (9, 0)
E      0    (9, 0)
E      0    (9, 0)
E   dtype: object
E   object
```